src/main/java/com/xeeva/navigation/MainNav.java

```diff
@@ -1,53 +1,313 @@
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void verifyCartItem(String ItemDescription, String UnitPrice,
+			String UOM, String Quantity) {
+
+		// clicking cart link
+		clickCartItemsLink();
+
+		// Reading Rows/Columns Count from Cart Item grid
+		List<WebElement> cartItemRows = driver.findWebElements(By.xpath(".//*[@id='tblCartInfo']/tbody/tr"));
+		TestReporter.logStep("Total Rows available in Cart :"+ cartItemRows.size());
+		List<WebElement> cartItemCols = driver.findElements(By.xpath(".//*[@id='tblCartInfo']/tbody/tr["
+				+ (cartItemRows.size() - 1) + "]/td"));
+		TestReporter.logStep("Total Columns available in Cart :"+ cartItemCols.size());
+
+		// Verification - Smart form Item Updated with Quantity, UOM and UnitPrice
+		verify_UpdateCart(UnitPrice, UOM, Quantity, cartItemRows.size());
+
+	}
+
+	/**
+	 * @summary This Method performs all Verifications For  Local,Global,Favorite,ComparisonSceen Cart Items
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public void perform_CartItemVerifications(String UnitPrice,String UOM,String Quantity,String itemNumber) {
+		// clicking cart link
+		clickCartItemsLink();
+
+		// Reading Rows Count from CartItem grid
+		List<WebElement> cartItemRows = driver.findWebElements(By
+				.xpath(".//*[@id='tblCartInfo']/tbody/tr"));
+		TestReporter.logStep("Total Rows available in Cart :"+ cartItemRows.size());
+
+		// Click on Edit
+		click_CartItemEdit(itemNumber);
+
+		// Update - Unit Price , UOM
+		TestReporter.logStep("Item Qty befor Update : "+ txtItemQuantity.getAttribute("value"));
+		String actualQty = txtItemQuantity.getAttribute("value");
+
+		Sleeper.sleep(3000);
+		txtItemUnitPrice.isDisplayed();
+		txtItemUnitPrice.clear();
+		txtItemUnitPrice.safeSet(UnitPrice);
+		lstItemunitOfMeasure.select(UOM);
+
+		btnAddToCart.jsClick();
+
+		// clicking cart link
+		clickCartItemsLink();
+
+		// Reading Quantity after changing UOM
+		String expectedQty = driver.findElement(
+				By.xpath(".//*[@id='tblCartInfo']/tbody/tr["
+						+ (cartItemRows.size() - 1) + "]/td[5]/input"))	.getAttribute("value");
+		TestReporter.logStep("Item Qty after Update : "+ expectedQty);
+
+		// Click on Edit
+		click_CartItemEdit(itemNumber);
+
+		String expectedUP = txtItemUnitPrice.getAttribute("value");
+		String expectedUOM = lstItemunitOfMeasure.getFirstSelectedOption().getText();
+
+
+		// Verify -  Quantity changed as per UOM
+		TestReporter.assertTrue(!actualQty.equalsIgnoreCase(expectedQty),"Quantity Updated !!");
+		// Verify - updated UOM
+		TestReporter.assertTrue(UOM.equalsIgnoreCase(expectedUOM),"Unit of Measure Updated !!");
+		// Verify - Updated UP
+		TestReporter.assertTrue(UnitPrice.equalsIgnoreCase(expectedUP),"Unit Price Updated !!");
+
+
+	}
+
+
+	/**
+	 * @summary Verify Cart Items 
+	 * @author Lalitha Banda
+	 * @date 14/9/16
+	 **/
+	public boolean verifyCartValue(String ItemType){
+		boolean statsuFlag = false;
+		lblCartValue.isDisplayed();

```

Comment:
These 2 lines of code .isDisplayed() and .isEnabled() are not serving a purpose.  These methods do not wait for either of those conditions, all they do is return a boolean.  so If you are not doing anything with that boolean value, then you need to remove.  Line 259 & 260